bokeh/crossfilter/models.py

```diff
@@ -128,84 +219,174 @@ def set_input_selector(self):
         )
 
     def update_plot_choices(self, input_dict):
-        for k,v in input_dict.items():
+        """Sets object attributes corresponding to input_dict's values.
+
+        Args:
+          input_dict (dict): dict with x, y, and plot_type keys
+
+        """
+        for k, v in input_dict.items():
             if getattr(self, k) is None:
                 setattr(self, k, v)
 
     def column_descriptor_dict(self):
+        """Creates column stats dict with keys of column names.
+
+        Returns:
+          dict: dict with key per column in data, where values are column stats
+
+        """
         column_descriptors = {}
         for x in self.columns:
             column_descriptors[x['name']] = x
         return column_descriptors
 
     def continuous_columns(self):
+        """Returns list of column descriptors for the non-Discrete columns.
+
+        Returns:
+          list(dict): list of dicts, containing metadata about columns
+
+        """
         return [x for x in self.columns if x['type'] != 'DiscreteColumn']
 
     def discrete_columns(self):
+        """Returns list of column descriptors for the Discrete columns.
+
+        Returns:
+          list(dict): list of dicts, containing metadata about columns
+
+        """
         return [x for x in self.columns if x['type'] == 'DiscreteColumn']
 
     def make_plot_choices(self):
+        """Selects first two continuous columns for x,y during initial setup
+
+        Returns:
+          dict: x, y, and plot_type keys and values for initial setup
+
+        """
         x, y = [x['name'] for x in self.continuous_columns()[:2]]
-        return {'x' : x, 'y' : y, 'plot_type' : scatter}
+        return {'x': x, 'y': y, 'plot_type': 'scatter'}
 
     def set_plot(self):
-        if self.x == self.y:
-            return
+        """Makes and sets the plot based on the current configuration of app."""
+
         plot = self.make_plot()
         self.plot = plot
         curdoc()._add_all()
 
     def make_plot(self):
-        if all([len(self.facet_x) ==0,
+        """Makes the correct plot layout type, based on app's current config."""
+
+        # no faceting
+        if all([len(self.facet_x) == 0,
                 len(self.facet_y) == 0,
                 len(self.facet_tab) == 0]):
             return self.make_single_plot()
 
-        if all([len(self.facet_x) !=0,
-                len(self.facet_y) == 0,
+        # x xor y faceting
+        if all([(len(self.facet_x) != 0) ^ (len(self.facet_y) != 0),
```

Comment:
I would probably use `if all([(len(self.facet_x) != 0) != (len(self.facet_y) != 0),` but the bitwise is valid too...